### ISSUE TITLE
(Option) unique SKU 2

### ISSUE THREAD

**Issue by zehawki** [vendure-ecommerce/vendure#2328]
**Is your feature request related to a problem? Please describe.**
I get why https://github.com/vendure-ecommerce/vendure/issues/854 cannot be done. But at least the variants of the same product should NOT be allowed to have the same SKU. These variants will come from the same vendor.

**Describe the solution you'd like**
Consider forcing unique SKU per variant.

**Additional context**
Subset of https://github.com/vendure-ecommerce/vendure/issues/854

**Comment by michaelbromley** [comment]
Yes, this is reasonable. We would need a application-level check for the create & update mutations that checks the existing SKUs of the other variants in the current product.

**Comment by dlhck** [comment]
This feature request has been moved to GitHub Discussions for community voting.
Please upvote the discussion if you would like to see this implemented: https://github.com/vendurehq/vendure/discussions/4286
